kv: silence spammy log

tbg opened the pull request:
This can fill pages of logs when a cluster starts, especially when it
never manages to connect go Gossip.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/29140)
<!-- Reviewable:end -->

a-robinson commented on pkg/kv/dist_sender.go:
```diff
@@ -342,8 +342,10 @@ func (ds *DistSender) getNodeDescriptor() *roachpb.NodeDescriptor {
 		}
 	}
 	ctx := ds.AnnotateCtx(context.TODO())
```
Move this inside the conditional?

tbg committed:
kv: silence spammy log

This can fill pages of logs when a cluster starts, especially when it
never manages to connect go Gossip.

Release note: None
